packages/emitter-framework/test/typescript/components/value-expression.test.tsx

```diff
@@ -0,0 +1,178 @@
+import { Output, render } from "@alloy-js/core";
+import { SourceFile } from "@alloy-js/typescript";
+import { ArrayValue, EnumValue, Namespace, Value } from "@typespec/compiler";
+import { $ } from "@typespec/compiler/experimental/typekit";
+import { format } from "prettier";
+import { assert, describe, expect, it } from "vitest";
+import { ValueExpression } from "../../../src/typescript/components/value-expression.js";
+import { getProgram } from "../test-host.js";
+
+describe("ValueExpression", () => {
+  async function testValueExpression(value: Value, expected: string) {
+    const res = render(
+      <Output>
+        <SourceFile path="test.ts">
+          const val = <ValueExpression value={value} />
+        </SourceFile>
+      </Output>,
+    );
+    const testFile = res.contents.find((file) => file.path === "test.ts");
+
+    assert.exists(testFile, "test.ts file not rendered");
+    const actualContent = await format(testFile.contents as string, { parser: "typescript" });
+    const expectedContent = await format(`const val = ${expected}`, {
+      parser: "typescript",
+    });
+    expect(actualContent).toBe(expectedContent);
+  }
+
+  it("handles string value", async () => {
+    // Error: Default typekits may not be used until a program is set in the compiler.
+    // TODO: better way to set this up so I can use typekit?
+    await getProgram(``);
+    const value = $.value.createString("test");
+
+    await testValueExpression(value, `"test"`);
+  });
+
+  it("handles numeric value", async () => {
+    await getProgram(``);
+    const value = $.value.createNumeric(42);
+
+    await testValueExpression(value, `42`);
+  });
+
+  it("handles decimal numeric value", async () => {
+    await getProgram(``);
+    const value = $.value.createNumeric(42.5);
+    await testValueExpression(value, `42.5`);
+  });
+
+  it("handles boolean value", async () => {
+    await getProgram(``);
+    const value = $.value.createBoolean(true);
+    await testValueExpression(value, `true`);
+  });
+
+  it("handles null value", async () => {
+    await getProgram(``);
+    const value: Value = {
+      entityKind: "Value",
+      valueKind: "NullValue",
+      value: null,
+    } as Value;
+    await testValueExpression(value, `null`);
+  });
+
+  it("handles empty array value", async () => {
+    // TODO: is there a better way to create an empty array value?
```

Comment:
like some typekit? If not I might use the `@example` pattern like in https://github.com/microsoft/typespec/pull/6972/files#diff-df99d4d0d5d18ddbfb74cda34ed637f99c57829ef71d718ea822d7d3f110e355R112

Reply:
TypeKits to construct TS values should be added. Added here: https://github.com/microsoft/typespec/issues/6976